core/src/smartcontracts/isi/query.rs

```diff
@@ -567,10 +565,12 @@ mod tests {
         let tx_limits = &state_block.transaction_executor().transaction_limits;
         let va_tx = AcceptedTransaction::accept(tx, &chain_id, tx_limits)?;
 
-        let topology = Topology::new(UniqueVec::new());
+        let (peer_public_key, _) = KeyPair::random().into_parts();
+        let peer_id = PeerId::new("127.0.0.1:8080".parse().unwrap(), peer_public_key);
+        let topology = Topology::new(vec![peer_id]);
```

Comment:
I didn't think about that though. I didn't strictly impose the limit that there must be 1 peer in the topology, rather it's just that new topology has to be created with 1 peer at least. Do you agree with this change? Do you think we should impose a limit to prevent topology from going to 0?

Reply:
>  Do you think we should impose a limit to prevent topology from going to 0?

Right now we would panic in `Topology::new` if topology is empty. 
As well with calls to `leader` and `proxy_tail`.
But it's possible to get empty topology through `block_committed` call afaik.

So i think we should bring consistency.
Essentially empty topology is possible but it's the end of the blockchain since no new blocks would be produced ever.

Reply:
Found ticket #3522 created by my a while ago